Sources/Testing/Issues/Issue.swift

```diff
@@ -179,3 +179,217 @@ extension Issue.Kind: CustomStringConvertible {
     }
   }
 }
+
+// MARK: - Codable
+
+extension Issue {
+  /// A serializable type describing a failure or warning which occurred during a test.
+  @_spi(ExperimentalSnapshotting)
+  public struct Snapshot: Sendable, Codable {
+    /// The kind of issue this value represents.
+    public var kind: Kind.Snapshot
+
+    /// Any comments provided by the developer and associated with this issue.
+    ///
+    /// If no comment was supplied when the issue occurred, the value of this
+    /// property is the empty array.
+    public var comments: [Comment]
+
+    /// A ``SourceContext`` indicating where and how this issue occurred.
+    public var sourceContext: SourceContext
+
+    /// Whether or not this issue is known to occur.
+    public var isKnown = false
+
+    /// Initialize an issue instance with the specified details.
+    ///
+    ///
+    init(issue: Issue) {
```

Comment:
Simple casting initializers like this one should have unlabelled arguments, so: `init(_ issue: Issue)`.

Or, for this and the `Kind.Snapshot` initializer, we could use `snapshotting` as the label so that it's clear it's a destructive conversion. That might be better?